python/dask_cudf/dask_cudf/expr/_expr.py

```diff
@@ -121,3 +123,54 @@ def _patched_var(
 
 
 Expr.var = _patched_var
+
+
+# Add custom code path for RepartitionQuantiles, because
+# upstream logic fails when null values are present. Note
+# that the cudf-specific code path can also be used for
+# multi-column divisions in the future.
+
+
+def _quantile(a, q):
+    if a.empty:
+        # Avoid calling `quantile` on empty data
+        return None, 0
+    a = a.to_frame() if a.ndim == 1 else a
+    return (
+        a.quantile(q=q.tolist(), interpolation="nearest", method="table"),
+        len(a),
+    )
+
+
+def merge_quantiles(finalq, qs, vals):
+    from dask_cudf.sorting import merge_quantiles as mq
+
+    return mq(finalq, qs, vals).iloc[:, 0].to_pandas()
+
+
+_original_layer = RepartitionQuantiles._layer
+
+
+def _cudf_layer(self):
+    if hasattr(self._meta, "to_pandas"):
```

Comment:
Could you add a short comment about what exactly this code does differently from the upstream version such that it can handle nulls?

Reply:
>nit: this canary for whether or not we have a cudf-backed object is also true for pyarrow objects I think. I know this is a hypothetical right now, but just noting it.

Yeah, sorry. This was used as a simple toggle when I was experimenting in the dask-expr code itself. If the code lives in dask-cudf, we can do an instance check.

>Could you add a short comment about what exactly this code does differently from the upstream version such that it can handle nulls?

Yes. I agree that a more-thorough comment is definitely needed here.